### PR TITLE
Replace log facility

### PR DESCRIPTION
**Pull request by blocknotes**
We could improve the logging feature using a standard log interface.

### CODE REVIEW

**Comment by blocknotes** [comment]
Some improvements exposed by @elia are reported [here](https://github.com/nebulab/solidus_importer/pull/13#pullrequestreview-370453071)

File: lib/solidus_importer/base_importer.rb [hunk 1]
```diff
@@ -2,8 +2,17 @@
...
+    attr_reader :options, :logger
+
     def initialize(options)
       @options = options
+      @logger ||= begin
+        if @options[:logger] && !@options[:logger].is_a?(Class)
+          @options[:logger]
+        else
+          (@options[:logger] || ::SolidusImporter::BaseLogger).new
+        end
```
Comment (elia) [inline lib/solidus_importer/base_importer.rb]: I think this can be a bit more linear/readable by reducing the refs to `@options[:logger]`, here's a possible way to do it:

```suggestion
        logger = @options[:logger] || ::SolidusImporter::BaseLogger
        logger = logger.new if logger.is_a?(Class)
        logger
```

File: lib/solidus_importer/base_importer.rb [hunk 2]
```diff
@@ -19,6 +28,10 @@ def before_import(initial_context)
       initial_context
     end
 
+    ##
+    # Defines a method called when a process row fails
+    def process_row_failure(_processor, context:, error:); end
```
Comment (elia) [inline lib/solidus_importer/base_importer.rb]: This method seems unrelated to the PR/commit, It's probably worth moving it to a different commit, briefly explaining why it's there.

File: lib/solidus_importer/base_logger.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module SolidusImporter
+  class BaseLogger
+    delegate :debug, :error, :fatal, :info, :unknown, :warn, to: :logger_engine
```
Comment (elia) [inline lib/solidus_importer/base_logger.rb]: `::Logger` can be used directly if we're just delegating those methods, I guess this is a leftover of a more complex implementation that is now gone.

File: lib/solidus_importer/process_row.rb [hunk 4]
```diff
@@ -12,15 +12,17 @@ def initialize(importer, row)
...
         begin
           processor.call(context)
-        rescue StandardError => e
+        rescue => e # rubocop:disable Style/RescueStandardError
+          @importer.process_row_failure(processor, context: context, error: e)
           context.merge!(success: false, messages: e.message) # rubocop:disable Performance/RedundantMerge
           break
         end
       end
+      @importer.logger.info ::SolidusImporter::Log.new(source: @row, context: context)
```
Comment (elia) [inline lib/solidus_importer/process_row.rb]: Passing an object instead of a string to the logger I think would be surprising for most users as the standard logger interface (#info, #warn, #error, etc.) is very well known.

File: lib/solidus_importer/base_importer.rb [hunk 5]
```diff
@@ -2,8 +2,17 @@
...
+
     def initialize(options)
       @options = options
+      @logger ||= begin
+        if @options[:logger] && !@options[:logger].is_a?(Class)
+          @options[:logger]
+        else
+          (@options[:logger] || ::SolidusImporter::BaseLogger).new
+        end
+      end
```
Comment (elia) [inline lib/solidus_importer/base_importer.rb]: Would also be good to give the logger a chance to interact with the context itself, so we might decide to pass it to `.new` or even go a step further and always use a callable that takes in the importer as a parameter, e.g.: 

```suggestion
      @logger = @options[:logger] || ->(context) { ::SolidusImporter::BaseLogger.new(context) }
```

and then prepare it in https://github.com/nebulab/solidus_importer/blob/cf0584b779eebce6b7fc675ccbe236e767873724/lib/solidus_importer/process_row.rb#L15

That means that we can initialize the importer with anything that can respond to logger methods and ignore the _context_ parameter if we want to.

E.g. passing a tagged Rails logger:

```rb
SolidusImporter::BaseImporter.new(
  …
  logger: ->(context) { Rails.logger.tagged(context[:row_id]) }
)
```
